types/d3-sankey/index.d.ts

```diff
@@ -348,7 +348,7 @@ export interface SankeyLayout<Data, N extends SankeyExtraProperties, L extends S
      *
      * @param compare Node comparison function.
      */
-    nodeSort(compare: (a: SankeyNode<N, L>, b: SankeyNode<N, L>) => number): this;
+    nodeSort(compare: (a: SankeyNode<N, L>, b: SankeyNode<N, L>) => number | undefined | null): this;
```

Comment:
Same sort of thing here - the parameter is a function or undefined or null - the function shouldn't return undefined or null if provided.  I think.